modules/core/src/main/scala/org/tessellation/domain/snapshot/rewards/Rewards.scala

```diff
@@ -1,15 +1,102 @@
 package org.tessellation.domain.snapshot.rewards
 
 import cats.data.NonEmptySet
+import cats.effect.Async
+import cats.implicits.toComposeOps
+import cats.syntax.functor._
 
 import scala.collection.immutable.SortedSet
 
+import org.tessellation.dag.snapshot.SnapshotOrdinal
 import org.tessellation.schema.ID.Id
-import org.tessellation.schema.transaction.RewardTransaction
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.{RewardTransaction, TransactionAmount}
+import org.tessellation.security.SecurityProvider
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric._
 
 object Rewards {
+  val epochs: PosInt = 4
+  val epochDurationInYears: PosDouble = 2.5
+  val epochRewards: PosDouble = 853333333.20
+  val normalizationFactor: PosDouble = 1e8
+  val softStaking = SoftStakingConfig(1, SnapshotOrdinal(1L))
+
+  private[rewards] def getRewardsPerSnapshot(
+    snapshotOrdinal: SnapshotOrdinal,
+    snapshotsPerMinute: NonNegInt
+  ): NonNegDouble =
+    getEpochNumber(snapshotOrdinal, snapshotsPerMinute) match {
```

Comment:
I think we cannot base epoch on snaphshot ordinal. I think we should have some kind of count-down to the next epoch stored in snapshot, as we don't know how many time-triggered snapshot's there will be. On each time-triggered snapshot we would decrement the count-down.

Reply:
Even if the time trigger would change at some point then we can still calculate new value to countdown from and reach next epoch at the same moment as we planned for before, so it seems like a good idea.